pkg/tbtc/dkg.go

```diff
@@ -296,14 +312,14 @@ func (de *dkgExecutor) generateSigningGroup(
 
 					// Set up the attempt timeout signal.
 					attemptCtx, cancelAttemptCtx := context.WithCancel(
-						loopCtx,
+						ctx,
 					)
 					go func() {
 						defer cancelAttemptCtx()
 
 						err := de.waitForBlockFn(
-							loopCtx,
-							attempt.startBlock+dkgAttemptMaxBlockDuration,
+							ctx,
+							attempt.startBlock+dkgAttemptMaximumProtocolBlocks,
```

Comment:
If we want to adjust something, we need to adjust the on-chain `resultSubmissionTimeout`. The client `dkgAttemptMaximumProtocolBlocks` was chosen empirically to cover the worst case of protocol computations.

Reply:
These are the numbers of DKG retries we need to execute for non-attributable protocol:
- 1 malicious member: 100 retries worst case, 50 retries average case. (P = (99 choose 99) / (100 choose 99) = 0.01)
- 2 malicious member: 4950 retries worst case, 2475 retries average case.
- 3 malicious member: 161700 retries worst case, 80850 retries average case.

This is quite a lot and I think we should agree that if there is even one malicious member in the group, it is better not to create this group at all because it will require multiple retries during the signing:
- 1 malicious member, 99 honest members, P = (99 choose 51) / (100 choose 51) = 0.49
- 2 malicious members, 98 honest members, P = (98 choose 51) / (100 choose 51) = 0.23757575757
- 3 malicious members, 97 honest members,  P = (97 choose 51) / (100 choose 51) = 0.11393939393

So let's assume we do not allow for any malicious behavior during DKG but we take into account inactivities after the announcement phase. If we allow for 5 retries of the protocol we need `5 x (1 + 5 + 150 + 5) + 15x100 = 2305` blocks, which is about 7.5h assuming 15 blocks delay for a result publication. What do you think?

Reply:
That makes sense!

Reply:
Created an issue to capture this work https://github.com/keep-network/keep-core/issues/3455